### PR TITLE
Add docstrings and tests for geograph discount feature

### PR DESCRIPTION
**Pull request by danialmalik**
### Story link:
https://edlyio.atlassian.net/browse/EDE-298

### description
* Restrict only "POST" method in the views
* Add :
    - Docstrings
    - Test cases
for the "Geography Based Discounts" feature

### CODE REVIEW

File: ecommerce/ucsd_features/constants.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+"""
+Constants used in UCSDFeatures app.
+"""
+
+# Email commtype codes
```
Comment (mumarkhan999) [inline ecommerce/ucsd_features/constants.py]: `commtype` Is it a typo?

File: ecommerce/ucsd_features/services/coupons.py [hunk 2]
```diff
@@ -20,12 +20,46 @@
...
+            category_slug (str): Slug of the category
+            options (dict): Extra keyword arguments
+
+            currently supported extra keyword arguments are:
+                only_multi_course_coupons (Boolean): if only Coupon Products which are of
+                                                    "Multiple Course" type should be returned
+
+        Returns:
+            Queryset<Product>: queryset containing the coupon Product(s)
+
```
Comment (mumarkhan999) [inline ecommerce/ucsd_features/services/coupons.py]: kindly remove this line.

File: ecommerce/ucsd_features/services/coupons.py [hunk 3]
```diff
@@ -20,12 +20,46 @@
...
```
Comment (mumarkhan999) [inline ecommerce/ucsd_features/services/coupons.py]: remove that line too.

File: ecommerce/ucsd_features/tests/test_utils.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+
+        return_value = send_email_notification(email=None, commtype_code='NOT_USED', context={})
+        self.assertFalse(return_value)
+        mocked_logger_error.assert_called_once_with(expected_error_message)
+
+    @patch('ecommerce.ucsd_features.utils.CommunicationEventType.objects.get_and_render', side_effect=Exception)
+    @patch('ecommerce.ucsd_features.utils.logger.error')
+    def test_send_email_notification_with_invalid_commtype_code_and_exception(self,
+                                                                              mocked_logger_error,
+                                                                              _):
```
Comment (mumarkhan999) [inline ecommerce/ucsd_features/tests/test_utils.py]:
```suggestion
    def test_send_email_notification_with_invalid_commtype_code_and_exception(self, mocked_logger_error, _):
```

File: ecommerce/ucsd_features/constants.py (hunk 1 above)
Comment (danialmalik) [reply ecommerce/ucsd_features/constants.py]: nope. edX uses this

File: ecommerce/ucsd_features/services/coupons.py [hunk 5]
```diff
@@ -63,6 +118,19 @@ def _is_coupon_valid_for_course(self, coupon, course_key, site=None):
         ])
 
     def get_available_vouchers(self, coupons):
+        """
+        Returns available vouchers derived from the provided coupon Products.
+        A voucher is considered to be availabe if:
+            - voucher is not assigned to anyone
+            - voucher's start date is in the past
+            - voucher's end date is in the future (voucher is not expired)
```
Comment (mumarkhan999) [inline ecommerce/ucsd_features/services/coupons.py]: Indent the other lists too in the comment section like you are doing here.

Comment (danialmalik) [reply ecommerce/ucsd_features/services/coupons.py]: done

**Pull request merged**